File: data/imported/knowledge/cells/typescript/merged/snippets/snippet_016.ts
async function testWorkbookProtection() {
  console.log("Testing workbook protection...");

  const workbook = new Workbook();
  workbook.protect(true, "password");

  await workbook.save("outputfiles/test_workbook_protection.xlsx");
  console.log("Saved to outputfiles/test_workbook_protection.xlsx");
}

File: data/imported/knowledge/cells/cpp/merged/snippets/snippet_016.cpp
TEST(CompatibilityTests, file_and_stream_roundtrip_preserve_data_validations)
{
    TempDir temp("compat-validations");
    const auto filePath = temp.Path("validations.xlsx");
    auto workbook = CreateValidationWorkbook();
    workbook.Save(filePath.string());

    std::vector<std::uint8_t> stream;
    workbook.Save(stream, SaveFormat::Xlsx);

    Workbook fromFile(filePath.string());
    Workbook fromStream(stream);
    AssertValidations(fromFile);
    AssertValidations(fromStream);
}